### DISCUSSION TITLE
Using LLMs to improve rnote

### DISCUSSION THREAD

**Discussion by tunjan** [flxzt/rnote#1332] [Development]
I would like to exploring the potential AI Generative models to improve rnote's codebase. These models have shown significant improvements in coding performance and context length recently (especially Google's Gemini), and they could prove useful to detect inefficiencies in the code. This would improve the usage of computing resources and increase mantainability. What are your thoughts on this?

**Comment by flxzt** [comment]
Unfortunately I am currently not interested in that.
This is a hobby project for me, and I have been having issues keeping up with reviewing merge requests already.
I'd rather spend my time reviewing and interacting with actual humans, and I doubt the maintainability will be increased by accepting AI generated code. At some point an architect is needed to keep the project steered in the right direction, and I don't see that AI can fulfill that role. Quite the contrary, it seems that (by looking at the first proposed change that was generated by AI) there is no deeper understanding of the correct functioning and even syntax by the AI. And if I request changes in a review round, there won't be anyone on the other side that is able to understand my reasoning, so I would imagine this would become frustrating quite quickly. It just seems to me that to accept AI PR's causes more headaches than what it would bring as benefit at this point in time.

**Comment by dkmar** [comment]
Lol I thought this was going to be about using AI to neaten handwriting, that would be sweet.
